primitives/core/src/hexdisplay.rs

```diff
@@ -53,6 +53,18 @@ impl<'a> sp_std::fmt::Debug for HexDisplay<'a> {
 	}
 }
 
+/// Extension trait for wrapping a [`HexDisplay`].
+pub trait HexDisplayExt {
+	/// Display self as hex string.
+	fn hex_display(&self) -> HexDisplay<'_>;
+}
+
+impl<T: AsBytesRef> HexDisplayExt for T {
+	fn hex_display(&self) -> HexDisplay<'_> {
+		HexDisplay::from(self)
+	}
+}
```

Comment:
I think this is not needed, HexDisplay has a method from, you can do:
```diff
@@ -302,7 +302,7 @@ impl Builder {
                                        "downloaded data for module {} (count: {} / prefix: {:?}).",
                                        f,
                                        module_kv.len(),
-                                       hashed_prefix.hex_display(),
+                                       HexDisplay::from(&hashed_prefix),
                                );
                                filtered_kv.extend(module_kv);
                        }
```

If we add too much generic implementation on `T` like this I think it makes doc less readable.

Alternatively you could implement, if you want to be able to call from also from generic :
```rust
impl<'a, T: AsBytesRef> From<&'a T> for HexDisplay<'a> {
	fn from(t: &'a T) -> Self {
		Self::from(t)
	}
}
```

Reply:
I think I am too biased in favour of using this type of extension traits as opposed to normal method :D but fair enough and will simply use `::from`